### ISSUE TITLE
"Validation failed: Supplier can't be blank" on update to v1.0

### ISSUE THREAD

**Issue by pmackay** [openfoodfoundation/openfoodnetwork#618]
Am getting

```
Validation failed: Supplier can't be blank/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/validations.rb:56:in `save!'
/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/attribute_methods/dirty.rb:33:in `save!'
/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:264:in `block in save!'
/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:313:in `block in with_transaction_returning_status'
/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/database_statements.rb:192:in `transaction'
/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:208:in `transaction'
/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:311:in `with_transaction_returning_status'
/home/ubuntu/.gem/ruby/1.9.1/gems/activerecord-3.2.21/lib/active_record/transactions.rb:264:in `save!'
/home/ubuntu/apps/openfoodnetwork/current/db/migrate/20141003060622_add_standard_variant_to_products.rb:20:in `block in up'
```

when running an update to v1.0 on UK staging.

**Comment by pmackay** [comment]
@aidandunsdon any idea if any of the Stroudco products could be defined without a supplier? 

cc @lin-d-hop in case you have any quick ideas.

**Comment by NickWeir63** [comment]
Could this be all the 'store cupboard' items which Stroudco sells?  These are bought in bulk from Essential (wholesaler) and sold as 'Stroudco' products - ie they dont have a separate producer.  We get round this by setting up Stroudco as a producer in the old system.  I am not sure how Aidan got round this with the migration

**Comment by aidandunsdon** [comment]
@pmackay @NickWeir63 @lin-d-hop   I will try to take a look at this later today. I did get issues with null values when merging staging and production databases but the nulls were in the data already on production not in Stroudco or DF. Maybe this was due to data entered into early versions of Fife foodhub etc on production before staging was created when nulls were tolerated. Will investigate further.

**Comment by aidandunsdon** [comment]
Some mandatory supplier_id fields missing in table spree_products fixing them with #621

**Comment by pmackay** [comment]
I think your changes resolved this issue with the update to 1.0. Have noted a different issue here https://github.com/openfoodfoundation/openfoodnetwork/issues/627#issuecomment-121757565.

**Comment by pmackay** [comment]
Staging and production updated to 1.0.
